### PR TITLE
refactor: cobra.io.json

### PR DESCRIPTION
**Pull request by synchon**
* [ ] fix #(issue number)
* [x] description of feature/fix
* [ ] tests added/passed
* [ ] add an entry to the [next release](../release-notes/next-release.md)

This PR adds typing annotations for cobra.io.json and its unit tests. It also upgrades the code style for them to maintain Python 3.6+ compatibility.

### CODE REVIEW

**Comment by synchon** [comment]
To be merged after #1134 is merged.

**Review by Midnighter — APPROVED** [review]

**Review by Midnighter — CHANGES_REQUESTED** [review]
```
  src/cobra/__init__.py:17: in <module>
      from cobra import io
  src/cobra/io/__init__.py:5: in <module>
      from cobra.io.json import from_json, load_json_model, save_json_model, to_json
  src/cobra/io/json.py:5: in <module>
      from ..dict import model_from_dict, model_to_dict
  E   ModuleNotFoundError: No module named 'cobra.dict'
```

File: src/cobra/io/json.py [hunk 1]
```diff
@@ -1,74 +1,80 @@
-# -*- coding: utf-8 -*-
+"""Provide functions for I/O in JSON format."""
 
-from __future__ import absolute_import
+from typing import TYPE_CHECKING, Any
 
-from cobra.io.dict import model_from_dict, model_to_dict
+from ..dict import model_from_dict, model_to_dict
```
Comment (Midnighter) [inline src/cobra/io/json.py]:
```suggestion
from .dict import model_from_dict, model_to_dict
```

Comment (synchon) [reply src/cobra/io/json.py]: Shoot! Missed it. Sending a fix now.

**Comment by synchon** [comment]
Checks were taking some time, hence rebased it on latest devel.

**Review by Midnighter — APPROVED** [review]

**Pull request merged**